src/search/operator_counting/delete_relaxation_constraints.h

```diff
@@ -0,0 +1,73 @@
+#ifndef OPERATOR_COUNTING_DELETE_RELAXATION_CONSTRAINTS_H
+#define OPERATOR_COUNTING_DELETE_RELAXATION_CONSTRAINTS_H
+
+#include  "constraint_generator.h"
+
+#include "../task_proxy.h"
+
+#include <memory>
+
+namespace lp {
+class LPVariable;
+}
+
+namespace options {
+class Options;
+}
+
+namespace operator_counting {
+class DeleteRelaxationConstraints : public ConstraintGenerator {
+    bool use_time_vars;
+    bool use_integer_vars;
+
+    /* [U_o] Is op part of the relaxed plan?
+       Binary, indexed with op.id */
+    std::vector<int> lp_var_id_op_used;
+
+    /* [R_f] Is fact <V,v> reached by the relaxed plan?
+       Binary, indexed with var.id, value */
+    std::vector<std::vector<int>> lp_var_id_fact_reached;
+
+    /* [F_{o,f}] Is o the first achiever of fact <V,v> in the relaxed plan?
+       Binary, indexed with op.id, var.id, value */
+    std::vector<std::vector<std::vector<int>>> lp_var_id_first_achiever;
+
+    /* [T_o] At what time is o used first?
+       {0, ..., |O|}, indexed with op.id */
+    std::vector<int> lp_var_id_op_time;
+
+    /* [T_f] At what time is <V,v> first achieved?
+       {0, ..., |O|}, indexed with var.id, value */
+    std::vector<std::vector<int>> lp_var_id_fact_time;
+
+    /* Indices of constraints that change in every state
+       Indexed with var.id, value */
+    std::vector<std::vector<int>> constraint_ids;
+
+    /* The state that is currently used for setting the bounds.
+       Remembering this makes it faster to unset the bounds when the state
+       changes. */
+    std::vector<FactProxy> last_state;
```

Comment:
I think we try to avoid storing the proxies, so I recommend using FactPair here instead.

Reply:
I changed it but it requires changing the lookup functions as well. The functions themselves are now nicer and use `[f.var][f.value]` instead of `[f.get_variable().get_id()][f.get_value()]`. The price for this is that  we now have to construct Fact pairs outside: `FactPair(var_id, value)` instead of `var.get_fact(value)`  and lots of `f.get_pair()` instead of `f`. I'm fine with it, though.

Reply:
Thanks, I like the style of the new code (using FactPair) more.